pkg/executor/core/executor.go

```diff
@@ -62,87 +73,114 @@ func (executor Executor) VerifySubject(ctx context.Context, verifyParameters e.V
 	return result, nil
 }
 
-func (executor Executor) GetVerifyRequestTimeout() time.Duration {
-	timeoutMilliSeconds := defaultVerifyRequestTimeoutMilliseconds
-	if executor.Config != nil && executor.Config.VerificationRequestTimeout != nil {
-		timeoutMilliSeconds = *executor.Config.VerificationRequestTimeout
+// verifySubjectForRegoPolicy verifies the subject with results.
+// And it also returns the decision based on the verifier results if required.
+func (executor Executor) verifySubjectForRegoPolicy(ctx context.Context, verifyParameters e.VerifyParameters) (types.VerifyResult, error) {
+	results, err := executor.verifySubjectInternalWithoutDecision(ctx, verifyParameters)
+	if err != nil {
+		return types.VerifyResult{}, err
 	}
-	return time.Duration(timeoutMilliSeconds) * time.Millisecond
+	// If it requires embedded Rego Policy Engine make the decision, execute
+	// OverallVerifyResult to evaluate the overall result based on the policy.
+	result := types.VerifyResult{VerifierReports: results}
+	if !featureflag.PassthroughMode.Enabled {
+		result.IsSuccess = executor.PolicyEnforcer.OverallVerifyResult(ctx, result.VerifierReports)
```

Comment:
we might want to add a note how passthroughMode affect the Rego policy verification path.